### ISSUE TITLE
Unify best practices for countDocuments() and estimatedDocumentCount()

### ISSUE THREAD

**Issue by vkarpov15** [Automattic/mongoose#6745]
Re: discussion from https://github.com/Automattic/mongoose/issues/6713, leaving a note to think about whether we should add a method that switches between `countDocuments()` and `estimatedDocumentCount()` based on whether `filter` is empty or not.

**Comment by ILoveYouDrZaius** [comment]
I agree with this issue, I don't understand why I have to switch between two functions.

**Comment by vkarpov15** [comment]
There's some more detail about why you would use one vs the other in the API docs:

* https://mongoosejs.com/docs/api.html#model_Model.estimatedDocumentCount
* https://mongoosejs.com/docs/api.html#model_Model.countDocuments

TLDR; `estimatedDocumentCount()` is faster when you're looking to count how many documents are in the entire collection, but might not be 100% accurate in a [sharded cluster](https://docs.mongodb.com/manual/sharding/). If you're using a standalone MongoDB instance and have less than 100k documents per collection, it doesn't make much difference whether you use `estimatedDocumentCount()` or `countDocuments()`. It can make a big difference with large collections though.

**Comment by xxyyzz2050** [comment]
use count(0)  instead of count({filter}) or countDocument({filter}) to run estimatedDocumentCount()

**Comment by libook** [comment]
Is it the best practice to use estimatedDocumentCount() when there is no query parameter?
If it is, it will be very helpful to switch between estimatedDocumentCount() and countDocuments() with a single method.

**Comment by vkarpov15** [comment]
@libook yes, you should use estimatedDocumentCount when there's no filter param. There may be some reasons to use countDocumenta() if you're using sharding, but I'm not certain.

**Comment by shuwenhe** [comment]
CountDocuments

**Comment by vkarpov15** [comment]
I'm going to close this one, `estimatedDocumentCount()` and `countDocuments()` have been separate for long enough and have different semantics so I don't think there's a good way to determine which one to use without having more understanding of what the user is trying to do.
